source/extensions/filters/http/ext_proc/processor_state.cc

```diff
@@ -251,30 +251,20 @@ absl::Status ProcessorState::handleBodyResponse(const BodyResponse& response) {
       clearWatermark();
       onFinishProcessorCall(Grpc::Status::Ok);
       should_continue = true;
-    } else if (callback_state_ == CallbackState::StreamedBodyCallback ||
-               callback_state_ == CallbackState::StreamedBodyCallbackFinishing) {
-      bool delivered_one = false;
+    } else if (callback_state_ == CallbackState::StreamedBodyCallback) {
       Buffer::OwnedImpl chunk_data;
-      while (auto queued_chunk = dequeueStreamingChunk(delivered_one, chunk_data)) {
-        // Loop through queue in case some of it is chunks that were never
-        // delivered because the processing mode changed.
```

Comment:
This while loop is removed since there will be no mode change during data processing. So, for STREAMED, mode, the gRPC response is always for the first chunk in the queue.